compiler/backend_inkwell/candy_rt/candy_rt.h

```diff
@@ -0,0 +1,72 @@
+#ifndef __CANDY_RT_H
+#define __CANDY_RT_H
+
+#define int128_t long long int
+
+typedef enum
+{
+    CANDY_TYPE_INT = 42,
+    CANDY_TYPE_TEXT,
+    CANDY_TYPE_TAG,
+    CANDY_TYPE_LIST,
+    CANDY_TYPE_STRUCT,
+    CANDY_TYPE_FUNCTION,
+} candy_type_t;
+
+typedef struct
+{
+    void *environment;
+    struct candy_value *(*function)(struct candy_value *);
+} candy_function_t;
+
+typedef struct
+{
+    struct candy_value **keys;
+    struct candy_value **values;
```

Comment:
Would this also work and get rid of `candy_value`?

```suggestion
    candy_value_t **keys;
    candy_value_t **values;
```

Reply:
This doesn't work because of how C resolves types. It's fine with `struct candy_value`, but the typedef `candy_value_t` does not exist yet at this point.